src/vs/workbench/contrib/inlineChat/browser/inlineChatDecorations.ts

```diff
@@ -145,6 +145,11 @@ export class InlineChatDecorationsContribution extends Disposable implements IEd
 }
 
 GutterActionsRegistry.registerGutterActionsGenerator(({ lineNumber, editor, accessor }, result) => {
+	const inlineChatService = accessor.get(IInlineChatService);
+	const noProviders = Iterable.isEmpty(inlineChatService.getAllProvider());
+	if (noProviders) {
+		return;
+	}
```

Comment:
I think the fix is okay for the recovery, but may be we should not register the generator at all if there are no chat providers? We could listen to chat provider registrations and only when there is one, we would register the generator. Otherwise, this function is run always and just returns early every time?

Reply:
So the fix for a candidate should be as safe & small as possible. The current fix I think fits this requirement. I think it's good to be merged.